File: 2022/Java/01/Part1.java
import java.io.File;
import java.io.FileNotFoundException;
import java.util.Scanner;

public class Part1 {
  public static void main(String args[]) {
    try {

      if (args.length < 1) {
        System.err.println("Usage: part1 <filename>");
        System.exit(1);
      }

      String filename = args[0];
      File f = new File(filename);
      Scanner fileReader = new Scanner(f);
      int lineTotal = 0;
      int max = 0;

      while(fileReader.hasNext()) {
        try {
          // If parsing was correct then add to line total;
          int line = Integer.parseInt(fileReader.nextLine());
          lineTotal += line;
        } catch (NumberFormatException nfe) {
          if (lineTotal > max) {
            max = lineTotal;
          }
          lineTotal = 0;
        }
      }

      fileReader.close();

      System.out.printf("Max: %d\n", max);
      if (filename.equals("test_input.txt")) {
        assert max == 24000 : "Incorrect Answer";
        System.out.println("Test Passed!");
      }
    } catch (FileNotFoundException fnfe) {
      System.out.printf("Error: %s\n", fnfe.getMessage());
    } catch (Exception e) {
      System.out.printf("There was a problem: %s\n", e.getMessage());
    }
  }
}

File: 2022/C/01/part1.c
#include <assert.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <errno.h>

#define MAX_SIZE 4096
#define TEST_INPUT_FILENAME "test_input.txt"
#define TEST_INPUT_ANSWER 24000

int main(int argc, char** argv) {
  if (argc < 2) {
    fprintf(stderr, "Usage: part1 <filename>\n");
    return 1;
  }

  FILE* fd = fopen(argv[1], "r");
  if (fd == NULL) {
    perror("Error");
    return 1;
  }

  char* input = malloc(MAX_SIZE);
  long total = 0;
  long max = 0;

  while(fgets(input, MAX_SIZE, fd)) {
    if (strcmp(input, "\n") != 0) {
      total += atoi(input);
    } else {
      if (total > max) {
        max = total;
      }
      total = 0;
    }
  }

  printf("Max: %li\n", max);

  free(input);
  fclose(fd);

  if (strcmp(argv[1], TEST_INPUT_FILENAME) == 0 ) {
    assert(max == TEST_INPUT_ANSWER);
    puts("Test Passed!");
  }

  return 0;
}
